crates/subspace-networking/src/node_runner.rs

```diff
@@ -135,9 +154,32 @@ impl NodeRunner {
     }
 
     async fn handle_peer_dialing(&mut self) {
-        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
         let local_peer_id = *self.swarm.local_peer_id();
+        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
 
+        // Handle reserved peers first.
+        if !self.reserved_peers.is_empty() {
```

Comment:
I'm not sure if we need this if. We won't even dial anyone if `reserved_peers.is_empty()`

Reply:
I would keep it:
- redundant conversion operations
- less logging messages
- code indentation make the whole feature separated that IMHO improves method readability